src/main/java/org/kamiblue/client/mixin/client/MixinMinecraft.java

```diff
@@ -84,5 +86,10 @@ public void init(CallbackInfo info) {
         }
     }
 
+    @Inject(method = "runTickMouse", at = @At(value = "INVOKE", target = "Lorg/lwjgl/input/Mouse;getEventButton()I"))
+    public void mouseTick(CallbackInfo info) {
+        KamiEventBus.INSTANCE.post(new MouseClickEvent(Mouse.getEventButton(), Mouse.getEventButtonState()));
+    }
```

Comment:
Is there no way to do this without a mixin?

Reply:
The issue with the forge event is that it comes with no information, this is the only way to get it as it happens as understood by Minecraft. The forge one you can't even get the button, or even if it was a button being down, there might be a lwjgl method, however I don't know it (or at least one that won't involve breaking Minecraft.)